### PR TITLE
Jeg 18

### PR DESCRIPTION
**Pull request by Mikul17**

### CODE REVIEW

File: __tests__/TopBar.test.tsx [hunk 1]
```diff
@@ -1,26 +1,25 @@
 import React from "react";
 import { fireEvent, render, screen, waitFor } from "@testing-library/react";
 import TopBar from "@/components/topBar/TopBar";
-import { alpha, colors } from "@mui/material";
 import NotificationButton from "@/components/topBar/NotificationButton";
 
 describe("TopBar", () => {
   it("Check if TopBar renders correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
     const pageElement = screen.getByText("Sample");
 
     expect(pageElement).toBeInTheDocument();
   });
 
   it("Check if ThemeButton renders correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
     const themeButton = screen.getByTestId("theme-button");
 
     expect(themeButton).toBeInTheDocument();
   });
 
   it("Check if ThemeButton changes theme correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 4]
```diff
@@ -40,7 +39,7 @@ describe("TopBar", () => {
   });
 
   it("Check if NotificationButton renders correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 5]
```diff
@@ -56,14 +55,14 @@ describe("TopBar", () => {
   });
 
   it("Check if UserDropdown renders correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 6]
```diff
@@ -56,14 +55,14 @@ describe("TopBar", () => {
...
     const userDropdownButton = screen.getByTestId("user-dropdown-button");
 
     expect(userDropdownButton).toBeInTheDocument();
   });
 
   it("Check if UserDropdown opens correctly", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 7]
```diff
@@ -84,7 +83,7 @@ describe("TopBar", () => {
   });
 
   it("Check if UserDropdown closes after clicking outside", () => {
-    render(<TopBar page="Sample" />);
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: __tests__/TopBar.test.tsx [hunk 8]
```diff
@@ -103,4 +102,24 @@ describe("TopBar", () => {
       expect(userDropdown).not.toBeVisible();
     });
   });
+
+  it("Check if UserDropdown closes after clicking on settings", () => {
+    render(<TopBar page="Sample" handleSettingsClick={function (): void {} } />);
```
Comment (majer565) [inline __tests__/TopBar.test.tsx]: use arrow function

File: components/settings/SettingsModal.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  return (
+    <Modal open={props.isOpen} onClose={props.onClose} data-testid="settings-modal">
+      <Box
+        sx={{
+          position: "absolute",
+          top: "50%",
+          left: "50%",
+          transform: "translate(-50%, -50%)",
+          width: "33rem",
+          //height: "36rem",
```
Comment (majer565) [inline components/settings/SettingsModal.tsx]: remove comment

File: components/settings/SettingsModal.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+            display: "flex",
+            justifyContent: "space-between",
+            alignItems: "center",
+            borderRadius: "2rem 2rem 0 0",
+            margin: "1rem 1rem 0.5rem 1rem",
+          }}
+        >
+          <Typography variant="h3" sx={{ marginLeft: "1rem" }}>
+            {" "}
+            Ustawienia{" "}
```
Comment (majer565) [inline components/settings/SettingsModal.tsx]: delete this - {" "}

File: components/settings/SettingsModal.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+        <Box
+          sx={{
+            display: "flex",
+            justifyContent: "space-between",
+            alignItems: "center",
+            borderRadius: "2rem 2rem 0 0",
+            margin: "1rem 1rem 0.5rem 1rem",
+          }}
+        >
+          <Typography variant="h3" sx={{ marginLeft: "1rem" }}>
```
Comment (majer565) [inline components/settings/SettingsModal.tsx]: fontWeight="600"

File: components/settings/SettingsModal.tsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        </Box>
+        <HorizontalDivider
+          bgColor={theme.palette.mode === "light" ? alpha(colors.black[500], 0.2) : alpha(colors.white[500], 0.2)}
+          variant={"middle"}
+        />
+        <Box
+          sx={{
+            display: "flex",
+            justifyContent: "center",
+            alignItems: "space-evenly",
```
Comment (majer565) [inline components/settings/SettingsModal.tsx:61]: does this affect anything?

File: __tests__/TopBar.test.tsx (hunk 1 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 2 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 3 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 4 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 5 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 6 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 7 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: __tests__/TopBar.test.tsx (hunk 8 above)
Comment (Mikul17) [reply __tests__/TopBar.test.tsx]: fixed

File: components/settings/SettingsModal.tsx (hunk 9 above)
Comment (Mikul17) [reply components/settings/SettingsModal.tsx]: fixed

File: components/settings/SettingsModal.tsx (hunk 10 above)
Comment (Mikul17) [reply components/settings/SettingsModal.tsx]: fixed

File: components/settings/SettingsModal.tsx (hunk 11 above)
Comment (Mikul17) [reply components/settings/SettingsModal.tsx]: fixed

File: components/settings/SettingsModal.tsx (hunk 12 above)
Comment (Mikul17) [reply components/settings/SettingsModal.tsx:61]: without this, textFields don't render correctly inside the modal

File: components/settings/SettingsModal.tsx (hunk 10 above)
Comment (majer565) [reply components/settings/SettingsModal.tsx]: 44L still has {" "}

Comment (Mikul17) [reply components/settings/SettingsModal.tsx]: fixed

**Pull request merged**